### DISCUSSION TITLE
vCenter custom attribute sync with existing fields in Netbox

### DISCUSSION THREAD

**Discussion by MrgampRus** [bb-Ricardo/netbox-sync#293] [Ideas]
Hello Ricardo,
It would be nice to implement a function to synchronize vm Custom Attributes with existing fields in Netbox.
For example, we use three fields Custom Attributes Customer, Owner and IT Service, currently
we sync the data with a PowerShell script https://github.com/jwegner89/netbox-utilities, I am this one
script upgraded and added synchronization Custom Attributes Customer -> Netbox vm Contacts, Custom Attributes Owner -> Netbox vm Tenant,
Custom Attributes IT Service -> Netbox vm Tag.
The fact is that the script https://github.com/jwegner89/netbox-utilities is not very good, it does
there are a lot of entries in the Change Log even if the data does not change, your script works very well, but I can't write in python and if you could help me a little, for example
to synchronize at least one field with a separate module, I would do the rest myself using an example.
What does not suit me with your Custom Attributes synchronization option?
The fact is that the team that prepares reports against the data in CustomFilds they
I think that it is more flexible (when creating reports) and it will be correct to store this data in existing fields.
![Capture](https://user-images.githubusercontent.com/122436953/231406312-0e586499-b94b-4c8b-b70a-f237ffd18680.JPG)

**Comment by bb-Ricardo** [comment]
Hey, I'm not sure if I understood it correctly. Can you describe in detail with an exact example of what youbtry to achive?

**Reply by MrgampRus** [reply]
For example, in my vCenter for each virtual machine, Custom Attributes are filled in this way:
![Netb1](https://user-images.githubusercontent.com/122436953/231429347-bb96bae1-713f-4861-8911-6760c12b413d.JPG)
I'd like to sync these with the NetBox Tenant, Contacts and Tag fields:
![Netb2](https://user-images.githubusercontent.com/122436953/231430115-eda6e895-f290-4dff-a7b5-9398e5d8a820.JPG)
Accordingly, I expect that by updating the data in vCenter, they will be updated in Netbox.

**Reply by MrgampRus** [reply]
There is another problem, in Netbox there are 20 contacts filled in this way:
Ivanov Petr Antonovich at vCenter name Ivanov P.A. so that the script does not add a new contact, I added
custom field for contacts and entered abbreviated names there, then the script first checks if there is such
abbreviated name in the custom field, if there is, then assign it to the virtual machine; if not, then first add a new contact,
but fills in his name as in vCentre di addon fills in a custom field for him. I implemented it in PowerShell

![Netb3](https://user-images.githubusercontent.com/122436953/231433050-3e9491df-f67b-4d27-8f21-5be5ebd617d7.JPG)

**Reply by bb-Ricardo** [reply]
did you read the description of all the config options?

**Reply by MrgampRus** [reply]
Yes, maybe I missed something?

**Reply by bb-Ricardo** [reply]
Especially the name matching would never work. I guess the only solutions would be
a) fork the project and add it to your fork
b) use your current power shell implementation

**Reply by MrgampRus** [reply]
Thanks for the hint, only now I realized that I can use your script together with the powershell script - (leaving only the synchronization functionality in it Custom Attributes), running two scripts in turn.

**Comment by MrgampRus** [comment]
Good afternoon.  Ricardo can you help implement this feature?

**Reply by bb-Ricardo** [reply]
Hi,

This is unfurtunatly not just one feature. I prefer generic aproaches and this is quite a specific request and to implement additional  resources.

None of this exists at the moment.

You would need to define mappings: custom attribute X should be NetBox resource Y

**Comment by bb-Ricardo** [comment]
This is an open source project which is maintained in my free time, which means everything is done best effort.
